File: files/CH-149.py
import pandas as pd

path = "CH-149 Extract From Text.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=6)

input['Extracted'] = input.iloc[:, 0].str.extract(r'(?<=[\(\[\{\*])(.*?)(?=[\)\]\}\*])', expand=False).fillna('')
input = input.drop(columns="Text")

print(input.equals(test)) # Test

File: files/100-199/149/CH-149.R
library(tidyverse)
library(readxl)

path = "files/CH-149 Extract From Text.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:D7")

result = input %>%
  mutate(Extracted = str_extract(Text, "(?<=[\\(\\[\\{\\*])(.*?)(?=[\\)\\]\\}\\*])")) 

all.equal(result$Extracted, test$Extracted, check.attributes = FALSE)
#> [1] TRUE